packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -1057,15 +1092,23 @@ abstract contract AssetManager is IERC721Receiver {
      * @return Whether the reward distribution is done at least once or not.
      */
     function _distributeReward() internal returns (bool) {
-        // TODO - Query the index from L2OutputOracle
-        uint256 outputIndex = _nextRewardOutputIndex;
+        uint256 outputIndex = L2_ORACLE.latestFinalizedOutputIndex() + 1;
+        uint256 latestOutputIndex = L2_ORACLE.latestOutputIndex();
+
+        if (!L2_ORACLE.VALIDATOR_POOL().isTerminated(outputIndex)) {
+            return false;
+        }
```

Comment:
I think this check is meaningless, since `L2OutputOracle` already checks if `ValidatorPool` is terminated before calling `afterSubmitL2Output`, isn't it?

Reply:
It is necessary because it checks if `ValidatorPool` is terminated with the `latestFinalizedOutputIndex`, not the output index which `L2OutputOracle` checks it with.

Reply:
Okay, I got it. 

There's another question: how is the reward distributed, which is submitted right before the output index that the transition to `ValidatorManager` happens? Suppose the index of the last output of `ValidatorPool` is `N`, and when the output for index at `N + 7 days` is submitted, it will call `_distributeReward()` and return `false`, so that the last output submitter for `ValidatorPool` cannot get reward. Am I missing something?

Reply:
Yes, the point what you said is to be done by manually before to distribute the first reward in `ValidatorManager`. In the description of this PR, I wrote that `For the outputs submitted last 7 days immediately before the upgrade, they should be unbonded manually.`. If the finalization period of the last output of `ValidatorPool` elapsed, we need to manually send `ValidatorPool.unbond()` transaction then `ValidatorManager` can distribute reward.

Reply:
Oh I got it, thanks.